ruby-call-signature/src/main/java/org/jetbrains/ruby/codeInsight/types/signature/serialization/MethodInfoSerialization.kt

```diff
@@ -58,3 +59,81 @@ fun Location(stream: DataInput): Location {
 
     return Location(path, lineno)
 }
+
+object SignatureInfoSerialization {
+
+    val PROTOCOL_VERSION = 1
+
+    fun serialize(signatureInfos: List<SignatureInfo>, stream: DataOutput) {
+        val gemInfo2Id = LinkedHashMap<GemInfo, Int>()
+        val classInfo2Id = LinkedHashMap<ClassInfo, Int>()
+        signatureInfos.forEach {
+            val classInfo = it.methodInfo.classInfo
+            val gemInfo = classInfo.gemInfo
+            classInfo2Id.putIfAbsent(classInfo, classInfo2Id.size)
+            gemInfo?.let {
+                gemInfo2Id.putIfAbsent(gemInfo, gemInfo2Id.size)
+            }
+        }
+        stream.writeInt(PROTOCOL_VERSION)
+        stream.writeInt(gemInfo2Id.size)
+        var iter = 0
```

Comment:
Not sure this assertion is very helpful

Reply:
I don't like the looks. The method is 40 lines long so every such variable complicates the state for the reader.